### PR TITLE
JDBC: Implement enough for CLion to be happy

### LINKED ISSUE #953: CLion JDBC support
At least org.duckdb.DuckDBConnection.getCatalog needs to be implemented for it to work:

```
[snip]
Caused by: java.sql.SQLFeatureNotSupportedException
        at org.duckdb.DuckDBConnection.getCatalog(DuckDBConnection.java:164)
        at com.intellij.database.remote.jdba.jdbc.JdbcIntermediateFacade.obtainConnectionInfoFromJdbc(JdbcIntermediateFacade.java:153)
        at com.intellij.database.remote.jdba.jdbc.JdbcIntermediateFacade.getConnectionInfo(JdbcIntermediateFacade.java:134)
        at com.intellij.database.remote.jdba.impl.RemoteFacadeImpl.getConnectionInfo(RemoteFacadeImpl.java:43)
[snip]
```

### PR DESCRIPTION
**Pull request by jdahlin**
This is enough to get basic SELECT/INSERT statements working in the query console.
Obviously a lot left to get introspection and so working.

Since it's really hard to debug JDBC drivers I had to add print statements before raising SQLFeatureNotSupportedException. I can remove them, but I think they'll be useful for someone who needs to do this kind of debugging in the future.

Fixes #953

### CODE REVIEW

File: tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java [hunk 1]
```diff
@@ -25,14 +25,17 @@
...
+    protected String catalog;
+    protected String schema;
 	public DuckDBConnection(DuckDBDatabase db) throws SQLException {
 		if (db.db_ref == null) {
 			throw new SQLException("Database was shutdown");
 		}
 		conn_ref = DuckDBNative.duckdb_jdbc_connect(db.db_ref);
 		DuckDBNative.duckdb_jdbc_set_auto_commit(conn_ref, true);
 		this.db = db;
+		this.catalog = "catalog";
```
Comment (hannesmuehleisen) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java:37]: Why are those constants "catalog"? I think they should not be initialised like so.

File: tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java [hunk 2]
```diff
@@ -111,6 +114,7 @@ public void clearWarnings() throws SQLException {
 
 	public void setTransactionIsolation(int level) throws SQLException {
 		if (level > TRANSACTION_REPEATABLE_READ) {
+	        System.out.format("DuckDBConnection.setTransactionIsolation %d\n", level);
```
Comment (hannesmuehleisen) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java:117]: Is that a leftover?

File: tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java [hunk 3]
```diff
@@ -121,6 +125,7 @@ public int getTransactionIsolation() throws SQLException {
 
 	public void setReadOnly(boolean readOnly) throws SQLException {
 		if (readOnly != db.read_only) {
+	        System.out.format("DuckDBConnection.setReadOnly %s\n", readOnly);
```
Comment (hannesmuehleisen) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBConnection.java:128]: Leftover as well?

**Review by hannes — COMMENTED** [review]
Thanks for this PR, however could you please comment on the `println` calls all over the place and the hard-coded "schema" and "catalog" values?

**Comment by hannesmuehleisen** [comment]
@jdahlin any update here?

**Comment by Mytherin** [comment]
Closing this issue because of lack of activity, feel free to re-open.

**Pull request closed without merging**